backend/dataall/api/Objects/DataPipeline/resolvers.py

```diff
@@ -79,6 +79,9 @@ def update_pipeline(context: Context, source, DataPipelineUri: str, input: dict
             data=input,
             check_perm=True,
         )
+    if (pipeline.template == ""):
+        stack_helper.deploy_stack(context, pipeline.DataPipelineUri)
+
```

Comment:
Nice, this is the part that avoid re-deployment of the pipeline for template-pipelines